### PR TITLE
Add time traveling world clock extension

### PR DESCRIPTION
**Pull request by markyfyi**
## Description

tl;dr it's a different take on a world clock to assist in coordinating meetings by allowing the the user to "scrub" backwards and forwards through time.  Answers the question "when it's [time] here, what time is it there?"

## Screencast

https://github.com/raycast/extensions/assets/1624703/e1bb46fc-f13f-47bb-b7a0-277a172bbf01

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by Ek2100** [comment]
Raycast does already do this built in however it is a bit finikey with what you have to type
image

**Comment by himarky** [comment]
I'm familiar and a big fan of the above functionality, however my extension aims to provide a view across multiple timezones.

Screenshot 2023-07-03 at 4 21 58 PM

**Comment by Ek2100** [comment]
That is true, this is actually really cool, now that i think about it. Have you considered adding a menu-bar command?

**Comment by himarky** [comment]
Thanks @Ek2100 ☺️ that could be an equally cool option, I'm just personally not a menu bar fan.

File: extensions/time-traveling-world-clock/src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+  if (h >= 6 && h < 10) {
+    return "icon-morning.png";
+  }
+
+  if (h >= 10 && h < 17) {
+    return "icon-day.png";
+  }
+
+  if (h >= 17 && h < 20) {
```
Comment (thomaslombart) [inline extensions/time-traveling-world-clock/src/index.tsx:200]: Maybe it's up for debate but I'd argue that `17` is still the day 😬 

Isn't there a way to know the sunrise/sunset hours for each city to return more accurate icons?

**Review by thomaslombart — CHANGES_REQUESTED** [review]
Congrats on the extension! I have some feedback on it.

## Action Icons

The actions could use icons to make them more scannable. For example, `+1hr` could have the `Icon.Plus` icon and be named `Add 1 hour`

CleanShot 2023-07-04 at 11 35 52@2x

## Search Placeholder

The search placeholder is meant to search among list results or filter them. It's a bit unintuitive to put instructions about actions in the search bar placeholder.

Maybe you could have some kind of onboarding by displaying a `List.Item` on top of the list view that can be dismissed. Something like that:

```tsx
<List.Item
  title="You can adjust hour with ⌘ + ← and ⌘ + →"
  icon={Icon.Info}
  actions={
    <ActionPanel>
      <Action icon={Icon.XMarkCircle} title="Dismiss Message" onAction={...} />
    </ActionPanel>
  }
/>
```

**Comment by himarky** [comment]
@thomaslombart thanks for the UX guidance. :)

I got rid of the placeholder, added some icons, and cleaned up the action names.  PTAL!

**Review by thomaslombart — COMMENTED** [review]
That's better. One last thing though: Could you add an empty list view with actions so that the user is able to add a city if they try to filter the current list?

CleanShot 2023-07-06 at 15 17 21@2x

Also, the placeholder could be a bit more detailed (e.g `Search cities` instead of just `Search`)

**Comment by himarky** [comment]
@thomaslombart love it.  Boom, done, PTAL

**Review by thomaslombart — APPROVED** [review]
